e2e/react-native/src/react-native-pcv3.test.ts

```diff
@@ -0,0 +1,79 @@
+import { ChildProcess } from 'child_process';
+import {
+  runCLI,
+  cleanupProject,
+  newProject,
+  uniq,
+  readJson,
+  runCommandUntil,
+  killProcessAndPorts,
+  fileExists,
+} from 'e2e/utils';
+
+describe('@nx/react-native/plugin', () => {
+  let project: string;
+  let appName: string;
+
+  beforeAll(() => {
+    project = newProject();
+    appName = uniq('app');
+    runCLI(
+      `generate @nx/react-native:app ${appName} --project-name-and-root-format=as-provided --no-interactive`,
+      { env: { NX_PCV3: 'true' } }
+    );
+  });
+
+  afterAll(() => cleanupProject());
+
+  it('nx.json should contain plugin configuration', () => {
+    const nxJson = readJson('nx.json');
+    const reactNativePlugin = nxJson.plugins.find(
+      (plugin) => plugin.plugin === '@nx/react-native/plugin'
+    );
+    expect(reactNativePlugin).toBeDefined();
+    expect(reactNativePlugin.options).toBeDefined();
+    expect(reactNativePlugin.options.bundleTargetName).toEqual('bundle');
+    expect(reactNativePlugin.options.startTargetName).toEqual('start');
+  });
+
+  it('should bundle the app', async () => {
+    const result = runCLI(
+      `bundle ${appName} --platform=ios --bundle-output=dist.js --entry-file=src/main.tsx`,
+      {
+        env: { NX_PCV3: 'true', NX_DAEMON: 'false' },
```

Comment:
I don't think you need to pass `NX_PCV3` or `NX_DAEMON` here since you already initialized it inside of your `beforeAll`